modules/videoio/src/videoio_registry.cpp

```diff
@@ -333,7 +333,24 @@ class VideoBackendRegistry
 
 } // namespace
 
-namespace videoio_registry {
+namespace videoio_registry { namespace internal {
+    const VideoBackendInfo& getBackendInfo(VideoCaptureAPIs api)
+    {
+        if (api == CAP_ANY)
+            CV_Error(Error::StsError, "Wrong backend ID: CAP_ANY");
+
+        const int N = sizeof(builtin_backends)/sizeof(builtin_backends[0]);
+        for (size_t i = 0; i < N; i++)
+        {
+            const VideoBackendInfo& info = builtin_backends[i];
+            if (info.id == api)
+                return info;
+        }
+
+        CV_Error(Error::StsError, "Unknown backend ID");
```

Comment:
This exception changes existed behavior of `.open()` method of VideoCapture for non-existed/disabled backends.

Perhaps we need to return pointer (`const VideoBackendInfo*`) and check for NULL in caller.

Reply:
Should I implement it like that?

Reply:
> This exception changes existed behavior of `.open()` method of VideoCapture for non-existed/disabled backends.
> 
> Perhaps we need to return pointer (`const VideoBackendInfo*`) and check for NULL in caller.

The open question is if it should be implemented like that.

Reply:
> @asmorkalov There is nothing to add from my side. No idea which updates are expected here.

>> This exception changes existed behavior of `.open()` method of VideoCapture for non-existed/disabled backends.
>> 
>> Perhaps we need to return pointer (`const VideoBackendInfo*`) and check for NULL in caller.

The open question is if it should be implemented like that.

Reply:
This is internal API, so it could be adopted later if you have better solution.

Anyway implementation should not raise exception through `CV_Error()` (as high-level API of `VideoCapture` doesn't do that in case of failure of opening process).
Changing that behavior would break a lot of existed user code. This is a compatibility question.